tests/test_runs/test_run.py

```diff
@@ -189,47 +197,105 @@ def test_to_from_filesystem_no_model(self):
         with self.assertRaises(ValueError, msg="Could not find model.pkl"):
             openml.runs.OpenMLRun.from_filesystem(cache_path)
 
+    @staticmethod
+    def assert_run_prediction_data(task, run, model):
+        # -- Get y_pred and y_true as it should be stored in the run
+        fold_map = np.full(int(task.get_dataset().qualities["NumberOfInstances"]), -1)
+        s_d = task.get_split_dimensions()
+        if (s_d[0] > 1) or (s_d[2] > 1):
+            raise ValueError("Test does not support this task type's split dimensions.")
+
+        for fold_id in range(s_d[1]):
+            _, test_indices = task.get_train_test_split_indices(repeat=0, fold=fold_id, sample=0)
+            fold_map[test_indices] = fold_id
+
+        X, y = task.get_X_and_y()
+
+        # Check correctness of y_ture and y_pred in run
+        for fold_id in range(s_d[1]):
+            # Get data for fold
```

Comment:
Why can't we simply read the masks from the splits?

Reply:
Which function would you use for that / are you referring to?

Reply:
Refactored loops to make this better